### DISCUSSION TITLE
CodeGen not working after upgrading project to .NET6 and LanguageExt 4.1.1

### DISCUSSION THREAD

**Discussion by MrYossu** [louthy/language-ext#1041] [General]
I have a project that uses CodeGen. Up until just now it was using .NET5 and LanguageExt 4.0.3, and was working fine. I have the following...

```c#
  [Free]
  public interface Fallible<T> {
    [Pure]
    T Success(T value);

    [Pure]
    string BadIdea(string reason);

    [Pure]
    Exception Failure(Exception ex);
  }
```

However, I just upgraded the project to .NET6 and LanguageExt 4.1.1, and now when I build, it doesn't seem to generate the code, as the extension class I have that provides helper methods for `Fallible` gives "_Cannot resolve symbol 'Failure'_" (same for the other two members).

In case it helps, here is the relevant section from the `.csproj` file. The only changes from the previous version are the version numbers mentioned above...

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <!-- Irrelevant bits snipped for clarity -->

  <ItemGroup>
    <PackageReference Include="LanguageExt.Core"
                      Version="4.1.1" />
    <PackageReference Include="LanguageExt.CodeGen"
                      Version="4.1.1"
                      PrivateAssets="all" />
    <PackageReference Include="CodeGeneration.Roslyn.BuildTime"
                      Version="0.6.1"
                      PrivateAssets="all" />
    <DotNetCliToolReference Include="dotnet-codegen"
                            Version="0.6.1" />
  </ItemGroup>
```

Anyone any idea what I need to do to get this working?

**Comment by Malgefor** [comment]
I had the same issue. Resolved it by removing all the bin/obj folders and first only building the project that contains the generated code, and after that the rest of the solution.

**Comment by MrYossu** [comment]
@Malgefor Well cut off both my legs and call me Shorty! That worked fine, thanks for the advice.
